### DISCUSSION TITLE
Modifying startup behavior

### DISCUSSION THREAD

**Discussion by sNiXx** [quarkusio/quarkus#43306] [Q&A] [answered]
Hello everyone,

I was wondering if there is a way to modify what is printed in the console on application startup?

Specifically what is printed after the application has started:
TestApp on JVM (powered by Quarkus 3.14.2) started in 5.120s. Listening on: http://localhost:8080

Or, alternatively, is there a listener that observes when the application has fully started?

Thank you!

**Comment by phillip-kruger** [comment]
https://quarkus.io/guides/lifecycle#listening-for-startup-and-shutdown-events

**Reply by sNiXx** [reply]
Yes, but this only explains the StartupEvent observer, at which point the application is not yet FULLY started.

**Comment by mkouba** [comment]
> I was wondering if there is a way to modify what is printed in the console on application startup?

AFAIK you can't modify the log message but you can customize the ASCII art banner - https://quarkus.io/guides/getting-started#banner.

> Or, alternatively, is there a listener that observes when the application has fully started?

As Phillip pointed out - you can define a CDI observer of the `StartupEvent`.

In Quarkus 3.14+, you can also execute some logic when the HTTP server is started: https://quarkus.io/guides/http-reference#how-to-execute-logic-when-http-server-started.

**Reply by sNiXx** [reply]
I am aware of the observer for the StartupEvent, but the application is only starting – it has not yet fully started. It can be observed from the logs that the application has fully started, so there must be an (internal) way of checking for this?

Thanks for the hint regarding the HTTP server, just recently added this to our startup logic.

If it is not possible for now, maybe this could be a feature request ☺️

**Reply by mkouba** [reply]
>  it has not yet fully started

Well, the problem is how would you define "fully started"?

**Reply by mkouba** [reply]
I mean extensions can do different things at startup. There is the `ServiceStartBuildItem` that can be produced by extensions and the `StartupEvent` is only fired after all services are started (or more precisely after all `@Record(RUNTIME_INIT)` recorder methods called from a build steps that produce `ServiceStartBuildItem` are executed). But it really depends on the logic executed by extensions.

**Reply by sNiXx** [reply]
Not quite sure, but there must already be some logic, because the log message I mentioned says "...started in...".

Either way, I am fine with what we currently have. I was just curious if the documentation was missing anything.

Thanks for all the great work on Quarkus – it really is a joy to work with!

**Reply by mkouba** [reply, marked as answer]
> Not quite sure, but there must already be some logic, because the log message I mentioned says "...started in...".

This message is logged right after all `RUNTIME_INIT` recorder methods are executed. And in most apps, the `StartupEvent` is fired shortly before this message; the HTTP server is usually started in between.

**Reply by mkouba** [reply]
> Thanks for all the great work on Quarkus – it really is a joy to work with!

:100: :+1:
